### ISSUE TITLE
[Compatabilty] add support for BepInEx GUI 

### ISSUE THREAD

**Issue by CatsArmy** [mattymatty97/LTC_AsyncLoggers#3]
BepInEx GUI is just another console i ported it to work outside of risk of rain 2 but it doesnt seem that it supports this mod for some reason i susspect something about the threads but im not sure

original mod isnt by me btw

**Comment by CatsArmy** [comment]
I did I'm trying to build your mod on my PC but I'm having alot of issues with the csproj file I'm trying to make it so when you schedule a delegate thingy that before after it logs the sender and the other param it uses the same system I am using for sockets and stuff and it also passes that info there

**Comment by CatsArmy** [comment]
BepInExLoggerPatcher
wrapLogs(object sender, LogEventArgs eventArgs) this is the function

If you wanna give it a shot yourself the socket Networking stuff should be in my git repo for Bepinex GUI in the BepInEx.GUI.Loader folder under the SendLogToClientSocket.cs

**Comment by CatsArmy** [comment]
I really don't think it would need to have a while true like the GUI is doing as you are doing it whenever something gets logged so if you make a static read-only int openport

And a static read-only IPaddress IP to create the socket it should work I tried to do it my self but your csproj file is really messy and confusing it feels as if it has a mind of it's own it just doesn't do what it needs to do

**Comment by mattymatty97** [comment]
i mean it just adds itself to the list of listeners
![image](https://github.com/mattymatty97/LTC_AsyncLoggers/assets/33039934/8aa5b2d6-307e-46e2-afe9-fd1ea6899d1a)
so there is no reason for the GUI not working

**Comment by CatsArmy** [comment]
Well yeah ik but you can download it via thunderstore and see for yourself I got no idea why it doesn't work to be honest

**Comment by mattymatty97** [comment]
i haven't been able to have BepInEx GUI work at all on Lethal Company sincerally 😅

**Comment by CatsArmy** [comment]
Really huh any errors?

**Comment by mattymatty97** [comment]
nothing... simply wont start

**Comment by CatsArmy** [comment]
Like the game won't start?

**Comment by CatsArmy** [comment]
> Like the game won't start? 

If so that's weird

**Comment by mattymatty97** [comment]
![image](https://github.com/mattymatty97/LTC_AsyncLoggers/assets/33039934/9e017e1e-6f60-400f-bda0-05d29ff1b3bc)
found the issue
seems GUI is targeting a wrong version of BepInEx

**Comment by mattymatty97** [comment]
![image](https://github.com/mattymatty97/LTC_AsyncLoggers/assets/33039934/7820b82b-13d5-4949-8456-d0eacfe9f1b3)

**Comment by CatsArmy** [comment]
I believe that the logs you can see are before it all gets patched by async logger

**Comment by CatsArmy** [comment]
Did putting the exe In patchers\CatsArmy-BepInEx_GUI
 fix the GUI not opening?

**Comment by mattymatty97** [comment]
mmk than this is the issue: 
`[14:00:56.9946058] [Error  :BepInEx.GUI.Loader] Error while trying to send log to socket: System.Net.Sockets.SocketException (0x80004005): An established connection was aborted by the software in your host machine.`

**Comment by CatsArmy** [comment]
Yep that explains it that's why I think you should also have the same methods I used to get the port and IP to then send the socket by your self

**Comment by mattymatty97** [comment]
but that's something i have no control over... you run the socket thing in a separate thread and enqueue things in a Queue
i'm not touching anything related to networking 🤔 
and no i'm not adding the socket stuff here... we/i need to figure out why me creating other threads is killing your socket

**Comment by mattymatty97** [comment]
seems to be caused by this somehow
https://github.com/mattymatty97/LTC_AsyncLoggers/blob/experimental/AsyncLoggers/src/Patches/BepInExLogEventArgsPatch.cs

**Comment by CatsArmy** [comment]
really huh

**Comment by CatsArmy** [comment]
really huh ill test that

**Comment by CatsArmy** [comment]
huh it does work why does it do that that makes no logical sence what so ever what

**Comment by mattymatty97** [comment]
i belive it's an issue with the https://github.com/CatsArmy/BepInEx.GUI/blob/main/BepInEx.GUI.Loader/src/LogPacket.cs
doing some weird stuff to convert the LogEventArgs that i modify

**Comment by CatsArmy** [comment]
> i belive it's an issue with the [CatsArmy/BepInEx.GUI@`main`/BepInEx.GUI.Loader/src/LogPacket.cs](https://github.com/CatsArmy/BepInEx.GUI/blob/main/BepInEx.GUI.Loader/src/LogPacket.cs?rgh-link-date=2024-03-28T14%3A18%3A15Z) doing some weird stuff to convert the LogEventArgs that i modify

well shit not i have to figure out the werid ass class no one understands shi

welp ill try and see whats going on

**Comment by mattymatty97** [comment]
i'm sorry 😂 at least we know where the issue is

**Comment by CatsArmy** [comment]
> i'm sorry 😂 at least we know where the issue is

yeah lol

**Comment by mattymatty97** [comment]
i was thinking this might also be an error on the receiving side of the GUI, as they have to do some parsing and i alter the format they might be trying to match

**Comment by CatsArmy** [comment]
No I don't think the GUI side of things is the issues here.
I think I'm somehow getting the return value before you patch perhaps and then it has an incorrect size for the string which breaks the packet tho I'm not sure

**Comment by mattymatty97** [comment]
have you figured out what's happening?

**Comment by CatsArmy** [comment]
> have you figured out what's happening?

Not sure yet haven't had the time to test and code a fix yet feel free to try and create a PR if you want to. I'm just busy next week I will have a bit more free time

**Comment by xiaoxiao921** [comment]
> i belive it's an issue with the https://github.com/CatsArmy/BepInEx.GUI/blob/main/BepInEx.GUI.Loader/src/LogPacket.cs doing some weird stuff to convert the LogEventArgs that i modify

If you modify the packet structure then you obviously have to fix the serializer + deserializer, can you link the modified packet structure when using timestamps?

**Comment by mattymatty97** [comment]
i'm not touching the packet as i do nlt reference anything about BepInEx GUI

but i do edit the output of LogEventArgs.ToString()

https://github.com/mattymatty97/LTC_AsyncLoggers/blob/preloader/AsyncLoggers%2Fsrc%2FPatches%2FBepInExLogEventArgsPatch.cs

**Comment by CatsArmy** [comment]
new error log this might help
```
Stack trace:
BepInEx.GUI.Loader.SendLogToClientSocket.SendPacketsToClientUntilConnectionIsClosed (System.Net.Sockets.Socket socket) (at ./SendLogToClientSocket.cs:96)
BepInEx.GUI.Loader.SendLogToClientSocket.<.ctor>b__10_0 () (at ./SendLogToClientSocket.cs:55)
System.Threading.ThreadHelper.ThreadStart_Context (System.Object state) (at <787acc3c9a4c471ba7d971300105af24>:0)
System.Threading.ExecutionContext.RunInternal (System.Threading.ExecutionContext executionContext, System.Threading.ContextCallback callback, System.Object state, System.Boolean preserveSyncCtx) (at <787acc3c9a4c471ba7d971300105af24>:0)
System.Threading.ExecutionContext.Run (System.Threading.ExecutionContext executionContext, System.Threading.ContextCallback callback, System.Object state, System.Boolean preserveSyncCtx) (at <787acc3c9a4c471ba7d971300105af24>:0)
System.Threading.ExecutionContext.Run (System.Threading.ExecutionContext executionContext, System.Threading.ContextCallback callback, System.Object state) (at <787acc3c9a4c471ba7d971300105af24>:0)
System.Threading.ThreadHelper.ThreadStart () (at <787acc3c9a4c471ba7d971300105af24>:0)
UnityEngine.Logger:LogException(Exception, Object)
AsyncLoggers.Wrappers.Unity.<>c__DisplayClass4_0:<LogException>b__0() (at ./src/Wrappers/Unity/AsyncLoggerWrapper.cs:36)
AsyncLoggers.Wrappers.Unity.<>c__DisplayClass28_0:<wrapCallback>b__0() (at ./src/Wrappers/Unity/AsyncLoggerWrapper.cs:136)
AsyncLoggers.Wrappers.JobWrapper:LogWorker() (at ./src/Wrappers/JobWrapper.cs:84)
AsyncLoggers.Wrappers.LogJob:Execute() (at ./src/Wrappers/JobWrapper.cs:68)
Unity.Jobs.JobStruct`1:Execute(LogJob&, IntPtr, IntPtr, JobRanges&, Int32)
```

**Comment by mattymatty97** [comment]
V2 ( now in Experimental release ) should now be compatible with the BepInEx GUI, do you mind testing?

**Comment by CatsArmy** [comment]
Wait what? I spent so long trying to fix that with another bug too that I eventually gave up from the lack of free time I had at the time sure I'll test it

**Comment by mattymatty97** [comment]
i recorded the entire program to use different logic, especially for the TimeStamps so now only the Listeners that request it will have the TimeStamps prepended.
